flink-runtime/src/test/java/org/apache/flink/runtime/failure/FailureEnricherUtilsTest.java

```diff
@@ -0,0 +1,275 @@
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.runtime.failure;
+
+import org.apache.flink.configuration.Configuration;
+import org.apache.flink.configuration.JobManagerOptions;
+import org.apache.flink.core.failure.FailureEnricher;
+import org.apache.flink.core.failure.FailureEnricherFactory;
+import org.apache.flink.core.plugin.PluginManager;
+import org.apache.flink.core.plugin.TestingPluginManager;
+import org.apache.flink.util.TestLoggerExtension;
+
+import org.apache.commons.collections.IteratorUtils;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Iterator;
+import java.util.Map;
+import java.util.Set;
+import java.util.concurrent.CompletableFuture;
+import java.util.stream.Collectors;
+
+import static org.apache.flink.core.testutils.FlinkAssertions.assertThatFuture;
+import static org.apache.flink.runtime.failure.FailureEnricherUtils.MERGE_EXCEPTION_MSG;
+import static org.assertj.core.api.Assertions.assertThat;
+
+/** Tests for the {@link FailureEnricherUtils} class. */
+@ExtendWith(TestLoggerExtension.class)
+class FailureEnricherUtilsTest {
+
+    @Test
+    public void testGetIncludedFailureEnrichers() {
+        Configuration conf = new Configuration();
+
+        // Disabled feature
+        conf.setString(JobManagerOptions.FAILURE_ENRICHERS_LIST, "");
+        Set<String> result = FailureEnricherUtils.getIncludedFailureEnrichers(conf);
+        assertThat(result).hasSize(0);
+
+        // Single enricher
+        conf.setString(JobManagerOptions.FAILURE_ENRICHERS_LIST, "enricher1");
+        result = FailureEnricherUtils.getIncludedFailureEnrichers(conf);
+        assertThat(result).hasSize(1);
+        assertThat(result).contains("enricher1");
+
+        // Multiple enrichers with spaces
+        conf.setString(JobManagerOptions.FAILURE_ENRICHERS_LIST, "enricher1, enricher2, enricher3");
+        result = FailureEnricherUtils.getIncludedFailureEnrichers(conf);
+        assertThat(result).hasSize(3);
+        assertThat(result).contains("enricher1", "enricher2", "enricher3");
+
+        // Multiple enrichers with spaces and empty values
+        conf.setString(
+                JobManagerOptions.FAILURE_ENRICHERS_LIST, "enricher1, ,enricher2,   enricher3");
+        result = FailureEnricherUtils.getIncludedFailureEnrichers(conf);
+        assertThat(result).hasSize(3);
+        assertThat(result).contains("enricher1", "enricher2", "enricher3");
+    }
```

Comment:
maybe add a test case for malformed lists such as using a different delimiter, like a period instead of a comma

Reply:
sure, added